src/structures/ChampionSkin.ts

```diff
@@ -83,7 +83,7 @@ export class ChampionSkin {
     this.vintageLoadingScreen = meraki.loadScreenVintagePath || undefined;
     this.internalId = data.id;
     this.chromas = meraki.chromas
-      .filter((c?: { id: number; name: string; chromaPath: string } | null) => c)
+      .filter((c?: { id: number; name: string; chromaPath: string } | undefined) => c)
```

Comment:
The `| null` is needed here, its because this is processing the raw data which might have `null`s here and needs to get rid of them.

Reply:
I'm actually thinking about what this code does? you are using a `filter` function on `meraki.chromas`. Then you specify each chroma as `c` and type it for `{ id: number; name: string; chromaPath: string }` which is useless since this is actually typed here:

https://github.com/TheDrone7/shieldbow/blob/ef96218cf8f830cc46aae83e7aa86abdf536626f/src/types/ChampionSkin.ts#L23-L27

then you are specifying that can be undefined by both `?` and ` | undefined` removing any of these won't change anything. Then you are returning `c` and I guess hope that javascript gods properly filter undefined by doing that 😆  My change doesn't really do anything there since types are stripped at runtime, so I can really write there whatever

I guess something clean that would work here is
```ts
.filter((c) => c !== undefined)
```
or for checking both undefined and nulls to make sure
```ts
.filter((c) => typeof c !== 'undefined')
```
or just keeping it as it without types, but I guess it's less readable that way
```ts
.filter((c) => c)
```
and typing this as
```ts
export interface MerakiSkinChroma {
  name: string;
  id: number;
  chromaPath: string;
}

export interface MerakiSkin {
  // [...]
  availability: string;
  chromas: (MerakiSkinChroma | undefined)[];
  lore: string;
```

Reply:
These can be both `undefined` and `null`, which is why checking for either is not plausible, creating a separate interface for the rest would probably be fine but did not seem necessary due to the fewer number of properties.

Reply:
Then just typeof check and `| undefined | null` for typings, not sure how to type this inline here

Reply:
Could do `.filter((c) => typeof c !== 'undefined' && typeof c !== 'null')`

Reply:
Or `.filter((c) => !['undefined', null'].includes(typeof c))` (jk, use the one before)

Reply:
![kekw](https://cdn3.emoji.gg/emojis/7529_KEKW.png)